android/java/org/chromium/chrome/browser/app/domain/TransactionsModel.java

```diff
@@ -0,0 +1,287 @@
+            BraveWalletService mBraveWalletService, AssetRatioService mAssetRatioService) {
+        synchronized (mLock) {
+            mContext = context;
+            this.mTxService = mTxService;
+            this.mKeyringService = mKeyringService;
+            this.mBlockchainRegistry = mBlockchainRegistry;
+            this.mJsonRpcService = mJsonRpcService;
+            this.mEthTxManagerProxy = mEthTxManagerProxy;
+            this.mSolanaTxManagerProxy = mSolanaTxManagerProxy;
+            this.mBraveWalletService = mBraveWalletService;
+            this.mAssetRatioService = mAssetRatioService;
+            addServiceObservers();
+        }
+    }
+
+    @Override
+    public void onNewUnapprovedTx(TransactionInfo txInfo) {
+        update(mActivityRef);
+    }
+
+    @Override
+    public void onUnapprovedTxUpdated(TransactionInfo txInfo) {
+        updateTx(txInfo, mActivityRef);
+    }
+
+    @Override
+    public void onTransactionStatusChanged(TransactionInfo txInfo) {
+        updateTx(txInfo, mActivityRef);
+    }
+
+    private void updateTx(
+            TransactionInfo txInfo, WeakReference<BraveWalletBaseActivity> mActivityRef) {
+        List<WalletListItemModel> items = _mParsedTransactions.getValue();
+        if (txInfo.txStatus == TransactionStatus.REJECTED
+                && items.stream().anyMatch(
+                        walletItem -> walletItem.getTransactionInfo().id.equals(txInfo.id))) {
+            // Remove rejected transaction
+            items = items.stream()
+                            .filter(walletItem
+                                    -> !walletItem.getTransactionInfo().id.equals(txInfo.id))
+                            .collect(Collectors.toList());
+            _mParsedTransactions.postValue(items);
+        } else if (txInfo.txStatus != TransactionStatus.UNAPPROVED) {
+            // Update status
+            WalletListItemModel walletItemTx =
+                    items.stream()
+                            .filter(walletItem
+                                    -> walletItem.getTransactionInfo().id.equals(txInfo.id))
+                            .findFirst()
+                            .orElse(null);
+            if (walletItemTx != null) {
+                walletItemTx.getTransactionInfo().txStatus = txInfo.txStatus;
+                Utils.updateWalletCoinTransactionStatus(
+                        walletItemTx, mContext, walletItemTx.getTransactionInfo());
+                _mParsedTransactions.postValue(items);
+            }
+        } else {
+            update(mActivityRef);
+        }
+    }
+
+    private static class AssetAccountsNetworkBalance {
+        HashMap<String, Double> assetPrices;
+        BlockchainToken[] userAssetsList;
+        HashMap<String, Double> nativeAssetsBalances;
+        HashMap<String, HashMap<String, Double>> blockchainTokensBalances;
+        NetworkInfo networkInfo;
+        List<AccountInfo> accountInfos;
```

Comment:
We can though don't think we have to be precise with English as same examples can be found in chromium and wallet. 
cc : @SergeyZhukovsky for suggestion

Reply:
It doesn't matter for me how do we name it as far as it makes sense for all of us. Let's change to `accountInfoList` in a new code we develop, as it looks like it will be better to understand for all devs. We will eventually change it everywhere when we touch old parts of code.

Reply:
updated: [a299fa3](https://github.com/brave/brave-core/pull/18631/commits/a299fa39de342cdcd2ff1be6ad8b150b107b1c04)